src/execution/interpreter_loop.rs

```diff
@@ -64,7 +70,20 @@ pub(super) fn run<H: HookSet>(
                 trace!("Instruction: NOP");
             }
             END => {
-                let maybe_return_address = stack.pop_stackframe();
+                // if this is not the very last instruction in the function
+                // just skip because it is a delimiter of a ctrl block
+
+                // TODO there is definitely a better to write this
+                let current_func_span = store
+                    .funcs
+                    .get(stack.current_stackframe().func_idx)
+                    .unwrap_validated()
+                    .code_expr;
+                if wasm.pc != current_func_span.from() + current_func_span.len() {
+                    continue;
+                }
```

Comment:
Is there a reason why we continue and we don't break?

```suggestion
                let current_func_span = store
                    .funcs
                    .get(stack.current_stackframe().func_idx)
                    .unwrap_validated()
                    .code_expr;

                if wasm.pc != current_func_span.from() + current_func_span.len() {
                    continue;
                }
```

Reply:
if this is not the very last END instr, END does not do anything, so we read the next instr (wasm.pc < last_end_idx). The dodgy case here is when the pc is erroneously after the very last END (wasm.pc > last_end_idx). I let it continue here, but we could also give an error or trap saying "instruction pointer is out of bounds"